carbon-design-system/carbon
# feat(link): refactor and add support for forwardRef

Closes #10898

This PR introduces support for `forwardRef` in the `Link` component. It also refactors `Link` in several ways, including:

- Update test suite to fully use RTL
- Update implementation for disabling a link to be more accessible
- Update `package.json` to remove the `window-or-global` package that is not needed

#### Changelog

**New**

**Changed**

- Update Link to support `forwardRef`
- Remove `window-or-global` dependency
- Update test suite for Link to use RTL

**Removed**

#### Testing / Reviewing

- Verify that the new tests correctly test `forwardRef` and that the element with `role="link"` is set to the ref
- Verify that no visual regressions have occurred, in particular for the disabled state
- Verify that this new implementation for disabling a link matches the accessibility notes linked in the component
